### PR TITLE
Implement Events

### PR DESCRIPTION
**Pull request by alinhle**
## What does this PR do?
- Introduce event emitter concept.
- Add `EventEmitterProvider` and a `withEmit` HOC that provides a `emit(eventName, value)` property.
- Integrate the GraphQL framework and redux into the event system
- Allow passing a `events` property into `Coral.Talk.render` options to support listening to events
```js
{
   // The argument passed is the event emitter from https://github.com/asyncly/EventEmitter2
   events: function(events) {
     events.onAny(function(eventName, data) {
       // logs all available events.
       console.log(eventName, data);
    });
  },
}

```
## How do I test this PR?

- Uncomment the example config in `views/article.ejs` and see the events flowing into the console :-)

### CODE REVIEW

**Review by okbel — APPROVED** [review]
This is 😍

File: client/coral-admin/src/index.js [hunk 1]
```diff
@@ -12,13 +14,19 @@ import 'react-mdl/extra/material.js';
 import './graphql';
 import {loadPluginsTranslations, injectPluginsReducers} from 'coral-framework/helpers/plugins';
 
+const eventEmitter = new EventEmitter();
+
+// TODO: pass redux actions through the emitter.
```
Comment (okbel) [inline client/coral-admin/src/index.js:19]: Is this TODO up to date?

Comment (alinhle) [reply client/coral-admin/src/index.js:19]: Yes, I'd love to add that after we are able to use the `store` service from `coral-framework` so I don't have to write duplicate code...

Comment (okbel) [reply client/coral-admin/src/index.js:19]: 👍

**Pull request merged**